universe/node/api_users.go

```diff
@@ -94,6 +95,163 @@ func (n *Node) apiUsersGetByID(c *gin.Context) {
+		user, err := n.LoadUser(userID)
+		if err != nil {
+			err := errors.WithMessage(err, "Node: apiUsersGetLatest: failed to get load user by id")
+			api.AbortRequest(c, http.StatusInternalServerError, "failed_to_load_user", err, n.log)
+			return
+		}
+
+		profile := user.GetProfile()
+
+		recent := dto.RecentUser{
+			ID:   user.GetID(),
+			Name: profile.Name,
+			Profile: dto.Profile{
+				Bio:         profile.Bio,
+				Location:    profile.Location,
+				AvatarHash:  profile.AvatarHash,
+				ProfileLink: profile.ProfileLink,
+			},
+		}
+
+		recents = append(recents, recent)
+	}
+
+	c.JSON(http.StatusOK, recents)
+}
+
+// @Summary Get owned worlds
+// @Schemes
+// @Description Returns a list of owned Worlds for a user
+// @Tags users
+// @Accept json
+// @Produce json
+// @Success 200 {array} dto.OwnedOdyssey
+// @Failure 500 {object} api.HTTPError
+// @Failure 400 {object} api.HTTPError
+// @Failure 404 {object} api.HTTPError
+// @Router /api/v4/users/{user_id}/worlds [get]
+func (n *Node) apiUsersGetOwnedWorlds(c *gin.Context) {
+	userID, err := umid.Parse(c.Param("userID"))
+	if err != nil {
+		err := errors.WithMessage(err, "Node: apiUsersGetOwnedWorlds: failed to parse user umid")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_user_id", err, n.log)
+		return
+	}
+
+	worlds := n.GetWorldsByOwnerID(userID)
+	ownedWorlds := make([]dto.OwnedWorld, 0, len(worlds))
+	for _, world := range worlds {
+		name := world.GetName()
+
+		ownedWorld := dto.OwnedWorld{
+			ID:      world.GetID(),
+			OwnerID: world.GetOwnerID(),
+			Name:    &name,
+		}
+
+		ownedWorlds = append(ownedWorlds, ownedWorld)
+	}
+
+	c.JSON(http.StatusOK, ownedWorlds)
+}
+
+// @Summary Search available users
+// @Schemes
+// @Description Returns user information based on a search query
+// @Tags users
+// @Accept json
+// @Produce json
+// @Success 200 {object} dto.UserSearchResult
+// @Failure 500 {object} api.HTTPError
+// @Failure 400 {object} api.HTTPError
+// @Failure 404 {object} api.HTTPError
+// @Router /api/v4/users/search [get]
+func (n *Node) apiUsersSearchUsers(c *gin.Context) {
+	type Query struct {
+		SearchQuery string `form:"query" binding:"required"`
+	}
+
+	inQuery := Query{}
+
+	if err := c.ShouldBindQuery(&inQuery); err != nil {
+		err := errors.WithMessage(err, "Node: apiUsersSearchUsers: failed to bind query")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_request_query", err, n.log)
+		return
+	}
+
+	users, err := n.apiUsersFilterUsers(inQuery.SearchQuery)
+	if err != nil {
+		err := errors.WithMessage(err, "Node: apiUsersSearchUsers: failed to filter objects")
+		api.AbortRequest(c, http.StatusBadRequest, "failed_to_filter", err, n.log)
+		return
+	}
+
+	c.JSON(http.StatusOK, users)
+}
+
+func (n *Node) apiUsersFilterUsers(searchQuery string) (dto.UserSearchResults, error) {
+	predicateFn := func(userID umid.UMID, user universe.User) bool {
+		var name string
+		loadedUser, _ := n.LoadUser(userID)
```

Comment:
`n.LoadUser` will execute SQL query each time, maybe we should consider specific SQL query to filter users by name